code/datums/martial/buster_style.dm

```diff
@@ -0,0 +1,479 @@
+/datum/martial_art/buster_style/proc/drop(mob/living/target)
+	for(var/atom/movable/K in thrown)
+		thrown.Remove(K)
+
+/datum/martial_art/buster_style/can_use(mob/living/carbon/human/H)
+	var/obj/item/bodypart/r_arm/robot/buster/R = H.get_bodypart(BODY_ZONE_R_ARM)
+	var/obj/item/bodypart/l_arm/robot/buster/L = H.get_bodypart(BODY_ZONE_L_ARM)
+	if(H.restrained() || H.get_active_held_item() || HAS_TRAIT(H, TRAIT_PACIFISM) || !(H.mobility_flags & MOBILITY_STAND))
+		for(var/atom/movable/K in thrown)
+			thrown.Remove(K)
+			walk(K,0)
+			K.density = old_density
+			animate(K, time = 0 SECONDS, pixel_y = 0)
+		return FALSE
+	if(L && R)
+		if((L?.bodypart_disabled) && (R?.bodypart_disabled))
+			to_chat(H, span_warning("The arms aren't in a functional state right now!"))
+			return FALSE
+		return TRUE //still got the other arm to pop off with
+	if(R || L)
+		if(R?.bodypart_disabled || L?.bodypart_disabled)
+			to_chat(H, span_warning("The [L ? "left" : "right"] buster arm isn't in a functional state right now!"))
+			return FALSE
+	return ..()
+
+
+/datum/martial_art/buster_style/proc/InterceptClickOn(mob/living/carbon/human/H, params, atom/target)
+	if(!(can_use(H)))
+		return
+	H.face_atom(target) //for the sake of moves that care about user orientation like mop and slam
+	if(H.a_intent == INTENT_DISARM)
+		mop(H,target)
+	if(H.a_intent == INTENT_HELP && (H==target))
+		arm_wire(H)
+	if(thrown.len > 0 && H.a_intent == INTENT_GRAB)
+		if(get_turf(target) != get_turf(H))
+			lob(H,target)
+	if(!H.Adjacent(target) || H==target)
+		return
+	if(H.a_intent == INTENT_HARM && isliving(target))
+		slam(H,target)
+	if(H.a_intent == INTENT_GRAB)
+		grapple(H,target)
+
+/datum/martial_art/buster_style/harm_act(mob/living/carbon/human/A, mob/living/D)
+	return TRUE // no punching plus slamming please
+
+
+/*---------------------------------------------------------------
+	start of wire section 
+---------------------------------------------------------------*/
+
+/datum/martial_art/buster_style/proc/arm_wire(mob/living/carbon/human/user)
+	for(var/obj/item/gun/magic/wire/J in user)
+		qdel(J)
+		to_chat(user, span_notice("The wire returns into your wrist."))
+		return
+	if(!COOLDOWN_FINISHED(src, next_wire))
+		to_chat(user, span_warning("You can't do that yet!"))
+		return
+	COOLDOWN_START(src, next_wire, COOLDOWN_WIRE)
+	var/obj/item/gun/magic/wire/gun = new /obj/item/gun/magic/wire (user)
+	user.put_in_hands(gun)
+
+/*---------------------------------------------------------------
+	end of wire section
+---------------------------------------------------------------*/
+/*---------------------------------------------------------------
+	start of grapple section
+---------------------------------------------------------------*/
+
+/datum/martial_art/buster_style/proc/grapple(mob/living/user, atom/target) //proc for picking something up to toss
+	var/turf/Z = get_turf(user)
+	target.add_fingerprint(user, FALSE)
+	if(next_grapple > world.time)
```

Comment:
this is not how cooldown defines work

Reply:
works on my server unless your definition of work doesn't mean "has a functioning cooldown"

Reply:
no i dont mean "works" i mean this is not using the cooldown define at all